helix-core/src/indent.rs

```diff
@@ -328,97 +622,28 @@ mod test {
     }
 
     #[test]
-    fn test_suggested_indent_for_line() {
-        let doc = Rope::from(
-            "
-use std::{
-    io::{self, stdout, Stdout, Write},
-    path::PathBuf,
-    sync::Arc,
-    time::Duration,
-}
-mod test {
-    fn hello_world() {
-        1 + 1;
-
-        let does_indentation_work = 1;
-
-        let test_function = function_with_param(this_param,
-            that_param
-        );
-
-        let test_function = function_with_param(
-            this_param,
-            that_param
-        );
-
-        let test_function = function_with_proper_indent(param1,
-            param2,
-        );
-
-        let selection = Selection::new(
-            changes
-                .clone()
-                .map(|(start, end, text): (usize, usize, Option<Tendril>)| {
-                    let len = text.map(|text| text.len()).unwrap() - 1; // minus newline
-                    let pos = start + len;
-                    Range::new(pos, pos)
-                })
-                .collect(),
-            0,
-        );
-
-        return;
+    fn test_treesitter_indent_rust() {
+        test_treesitter_indent("rust.rs", "source.rust");
     }
-}
-
-impl<A, D> MyTrait<A, D> for YourType
-where
-    A: TraitB + TraitC,
-    D: TraitE + TraitF,
-{
-
-}
-#[test]
-//
-match test {
-    Some(a) => 1,
-    None => {
-        unimplemented!()
+    #[test]
+    fn test_treesitter_indent_rust_2() {
+        test_treesitter_indent("indent.rs", "source.rust");
+        // TODO Use commands.rs as indentation test.
+        // Currently this fails because we can't align the parameters of a closure yet
+        // test_treesitter_indent("commands.rs", "source.rust");
     }
-}
-std::panic::set_hook(Box::new(move |info| {
-    hook(info);
-}));
-
-{ { {
-    1
-}}}
-
-pub fn change<I>(document: &Document, changes: I) -> Self
-where
-    I: IntoIterator<Item = Change> + ExactSizeIterator,
-{
-    [
-        1,
-        2,
-        3,
-    ];
-    (
-        1,
-        2
-    );
-    true
-}
-",
-        );
 
-        let doc = doc;
+    fn test_treesitter_indent(file_name: &str, lang_scope: &str) {
         use crate::diagnostic::Severity;
-        use crate::syntax::{
-            Configuration, IndentationConfiguration, LanguageConfiguration, Loader,
-        };
+        use crate::syntax::{Configuration, IndentationConfiguration, Loader};
         use once_cell::sync::OnceCell;
+        use std::path::PathBuf;
+        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        path.push("test-indent");
+        path.push(file_name);
```

Comment:
Let's put the fixtures under `helix-core/test/data/indent` or `helix-core/test/fixtures/indent`. The test cases themselves could also be under `test/`